main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Differ.scala

```diff
@@ -26,20 +26,77 @@ object Differ {
     case None =>
       // Case 1: No old AST. No difference.
       DeltaContext(Nil)
-    case Some(oldAst) => {
+    case Some(oldAst) =>
       // Case 2: We have an oldAst and a newAst. Compute their difference.
 
       // TODO: Add some comments and introduce helper functions.
-      val newDefs = (newAst.defs.keySet -- oldAst.defs.keySet).toList.map(sym => Delta.AddDef(sym, getCurrentTimestamp))
+      val newDefs = findModifiedDefs(oldAst, newAst)
       val newEnums = (newAst.enums.keySet -- oldAst.enums.keySet).toList.map(sym => Delta.AddEnum(sym, getCurrentTimestamp))
 
       DeltaContext(newDefs ++ newEnums)
-    }
+  }
+
+  /**
+    * Find all defs that has been modified.
+    *
+    * This is done by checking if a def in an oldAst is different from that same def in the newAst.
+    * If the def is modified, we add it to the list of deltas.
+    *
+    * @param oldAst the old AST.
+    * @param newAst the new AST.
+    * @return       a List[Delta] consisting of newly modified defs.
+    */
```

Comment:
Not quite sure how to make this even more precise? What do you think is missing?